kratos/linear_solvers/amgcl_solver.h

```diff
@@ -594,99 +478,308 @@ class AMGCLSolver : public LinearSolver< TSparseSpaceType,
 
                     ndof=1;
-                }
-                else
-                {
+                } else {
                     ndof++;
                 }
             }
         }
 
         if(old_ndof == -1)
-            mndof = 1;
+            mBlockSize = 1;
         else
-            mndof = ndof;
+            mBlockSize = ndof;
 
-        if(mverbosity > 1)
-        {
-            KRATOS_WATCH(mndof);
-        }
+        KRATOS_INFO_IF("AMGCL Linear Solver", mVerbosity > 1) << "mndof: " << mBlockSize << std::endl;
 
-        if(mprovide_coordinates == true)
-        {
-            mcoords.resize(rA.size1()/mndof);
+        if(mProvideCoordinates) {
+            mCoordinates.resize(rA.size1()/mBlockSize);
             unsigned int i=0;
-            for (ModelPart::DofsArrayType::iterator it = rdof_set.begin(); it!=rdof_set.end(); it+=mndof)
-            {
-                if(it->EquationId() < rA.size1() )
-                {
-                    ModelPart::NodesContainerType::iterator inode = r_model_part.Nodes().find(it->Id());
-                    mcoords[ i ] = inode->Coordinates();
+            for (auto it_dof = rDofSet.begin(); it_dof!=rDofSet.end(); it_dof+=mBlockSize) {
+                if(it_dof->EquationId() < rA.size1() ) {
+                    auto it_node = rModelPart.Nodes().find(it_dof->Id());
+                    mCoordinates[ i ] = it_node->Coordinates();
                     i++;
-
                 }
             }
         }
     }
 
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /**
+     * Print information about this object.
+     */
+    void  PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "AMGCL solver finished.";
+    }
+
+    /**
+     * Print object's data.
+     */
+    void  PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
 private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
 
-    double mTol;
-    unsigned int mmax_it;
-    int mverbosity;
-    int mndof;
-    unsigned int mgmres_size;
-    unsigned int mcoarse_enough;
-    bool mfallback_to_gmres;
-    bool mprovide_coordinates;
-    bool muse_block_matrices_if_possible;
-    std::vector<array_1d<double,3> > mcoords;
+    double mMaxTolerance;             /// The tolerance considered
```

Comment:
i would call it mTolerance (there is no reason for calling it MaxTolerance)

Reply:
I think this is just a minor aesthetically change, i prefer MaxTolerance :D